src/main/drivers/bus_i2c_hal.c

```diff
@@ -20,111 +20,114 @@
 
 #include <platform.h>
 
-#include "drivers/bus_i2c.h"
 #include "drivers/io.h"
+#include "drivers/io_impl.h"
 #include "drivers/nvic.h"
 #include "drivers/time.h"
+#include "drivers/rcc.h"
 
-#include "io_impl.h"
-#include "rcc.h"
-
+#include "drivers/bus_i2c.h"
+#include "drivers/bus_i2c_impl.h"
 
-#if !defined(SOFT_I2C) && defined(USE_I2C)
+#if defined(USE_I2C) && !defined(SOFT_I2C)
 
 #define CLOCKSPEED 800000    // i2c clockspeed 400kHz default (conform specs), 800kHz  and  1200kHz (Betaflight default)
 
 static void i2cUnstick(IO_t scl, IO_t sda);
 
-#if defined(USE_I2C_PULLUP)
-#define IOCFG_I2C IO_CONFIG(GPIO_MODE_AF_OD, GPIO_SPEED_FREQ_VERY_HIGH, GPIO_PULLUP)
-#else
-#define IOCFG_I2C IOCFG_AF_OD
-#endif
-
-#ifndef I2C1_SCL
-#define I2C1_SCL PB6
-#endif
-
-#ifndef I2C1_SDA
-#define I2C1_SDA PB7
-#endif
-
-#ifndef I2C2_SCL
-#define I2C2_SCL PB10
-#endif
-#ifndef I2C2_SDA
-#define I2C2_SDA PB11
+#define IOCFG_I2C_PU IO_CONFIG(GPIO_MODE_AF_OD, GPIO_SPEED_FREQ_VERY_HIGH, GPIO_PULLUP)
+#define IOCFG_I2C_NOPU IOCFG_AF_OD
```

Comment:
Yes. Make it consistent with IOCFG_I2C_PU. Also I'll change the name (NOPU beeing default value)